doc/context.md

```diff
@@ -0,0 +1,290 @@
+
+```js
+// Test the #shadowForm shadow DOM tree from the :root, skipping any <slot> elements
+axe.run({
+  include: {
+    fromShadowDom: ['#shadowForm', ':root']
+  },
+  exclude: {
+    fromShadowDom: ['#shadowForm', 'slot']
+  }
+});
+```
+
+## Combine Shadow DOM and Frame Context
+
+To select frames inside shadow DOM trees, or shadow DOM trees inside frames it is possible to use the a [`fromShadowDom`](#limit-shadow-dom-testing) as a selector in the [`fromFrames`](#limit-frame-testing) selector object. The following example shows how to test the `main` element, inside each `iframe`, that is part of the shadow DOM tree of `#appRoot`:
+
+```js
+await axe.run({
+  fromFrames: [
+    {
+      fromShadowDom: ['#appRoot', 'iframe']
+    },
+    'main'
+  ]
+});
+```
+
+These selectors can also be used on `include` or `exclude`. The following shows how to exclude the `footer`, as well as any `.commentBody` elements in the `#userComments` shadow DOM tree, inside the `#blog-comments` iframe:
+
+```js
+await axe.run({
+  exclude: [
+    'footer',
+    {
+      fromFrames: [
+        'iframe#blog-comments',
+        {
+          fromShadowDom: ['#userComments', '.commentBody']
+        }
+      ]
+    }
+  ]
+});
+```
+
+**Note**: Even though the iframe is inside the shadow DOM tree, the `fromShadowDom` selector object must be part of the `fromFrames` selector. Doing the reverse does not work and will cause an error.
+
+## Implicit Frame And Shadow DOM Selection
+
+In earlier versions of axe, using nested arrays was the only way to include or exclude elements inside iframes. These are still supported in axe, and is how axe works internally. This nested array syntax is used in the `target` property. For example, the following is a possible selector for a `.commentBody` element, in the shadow DOM tree of `#userComments`, inside the `#blog-comments` iframe:
+
+```js
+result = await axe.run();
+result.violations[0].nodes[0].target; // ['#blog-comments', ['#userComments', '.commentBody']]
+```
+
+To pass a `target` property into axe it must be wrapped in another array. This creates three nested arrays. The outer array to allow multiple selectors, the middle array to select into frames, and the inner array, which is optional, to select elements inside shadow DOM trees.
+
+While this syntax continues to be supported, it is recommended to avoid it, and to use the [fromFrames](#limit-frame-testing) and [fromShadowDom](#limit-shadow-dom-testing) object selectors instead, since these are more clearer and don't need to be wrapped in otherwise empty arrays.
```

Comment:
```suggestion
While this syntax continues to be supported, it is recommended to avoid it and to use the [fromFrames](#limit-frame-testing) and [fromShadowDom](#limit-shadow-dom-testing) object selectors instead since these are clearer and don't need to be wrapped in otherwise empty arrays.
```